packages/price-exporter/src/main.js

```diff
@@ -1,3 +1,270 @@
+      this.logger.info('Export operation completed successfully')
+    }
+  }
+
+  _preparePrices (flatPrices: Array<UnprocessedPriceObject>) {
+    return Promise.map(flatPrices, variantPrice => (
+      Promise.map(variantPrice.prices, individualPrice => (
+        this._resolveReferences('channel', individualPrice)
+        .then(price => this._resolveReferences('customerGroup', price))
+        .then(price => this._resolveReferences('custom', price))
+      ))
+      .then((prices: Array<ProcessedPriceObject>) => {
+        if (prices.length)
+          return prices.map(price => flatten(price))
+        return prices
+      })
+    ), { concurrency: 10 }) // Add currency to maximize cache usage
+  }
+
+  _resolveReferences (
+    type: string,
+    price: Object,
+  ): Promise<Object> {
+    // type: customerGroup, channel, custom
+    // Return the object if the type does not exist
+    if (!price[type])
+      return Promise.resolve(price)
+
+    const resolvedPrice: Object = Object.assign({}, price)
+    if (type === 'custom') {
+      delete resolvedPrice[type]
+      resolvedPrice.customField = price[type].fields
+
+      // Get from cache if it's already been cached
+      if (this._cache[price[type].type.id]) {
+        resolvedPrice.customType = this._cache[price.custom.type.id]
+        return Promise.resolve(resolvedPrice)
+      }
+
+      // If not in cache, resolve from API
+      const request = this._buildRequest(type, price)
+      return this.client.execute(request)
+      .then((response: Object): Promise<Object> => {
+        // Save resolved id to cache
+        this._cache[price[type].type.id] = response.body.key
+        resolvedPrice.customType = response.body.key
+        return Promise.resolve(resolvedPrice)
+      })
+    }
+    resolvedPrice[type] = {}
+
+    // Get from cache if it's already been cached
+    if (this._cache[price[type].id]) {
+      if (type === 'customerGroup')
+        resolvedPrice[type].groupName = this._cache[price[type].id]
+      else
+        resolvedPrice[type].key = this._cache[price[type].id]
+      return Promise.resolve(resolvedPrice)
+    }
+
+    // If not in cache, resolve from API
+    const request = this._buildRequest(type, price)
+    return this.client.execute(request)
+    .then((response: Object): Promise<Object> => {
+      // Save resolve to cache
+      if (type === 'customerGroup') {
+        resolvedPrice[type].groupName = response.body.name
+        this._cache[price[type].id] = response.body.name
+      } else {
+        this._cache[price[type].id] = response.body.key
+        resolvedPrice[type].key = response.body.key
+      }
+      return Promise.resolve(resolvedPrice)
+    })
+  }
+
+  _buildRequest (
+    type: string,
+    price?: Object,
+  ): ClientRequest {
+    const priceObject = { ...price }
+    let serviceType
+    let uri
+    switch (type) {
+      case 'custom':
+        serviceType = 'types'
```

Comment:
Would it work if we rename "custom" type to "types" and then just do this?
`serviceType = type`

Because this switch case looks complicated and maybe it is not needed in the end.